examples/notebooks/get-started/try-apache-beam-yaml.ipynb

```diff
@@ -0,0 +1,556 @@
+   ]
+  },
+  {
+   "metadata": {
+    "id": "GOOk81Jj_yUy",
+    "colab_type": "code",
+    "outputId": "d283dfb2-4f51-4fec-816b-f57b0cb9b71c",
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 170
+    }
+   },
+   "cell_type": "code",
+   "source": [
+    "# Run and print a shell command.\n",
+    "def run(cmd):\n",
+    "  print('>> {}'.format(cmd))\n",
+    "  !{cmd}\n",
+    "  print('')\n",
+    "\n",
+    "def save_to_file(content, file_name):\n",
+    "  with open(file_name, 'w') as f:\n",
+    "    f.write(content)\n",
+    "\n",
+    "# Install apache-beam.\n",
+    "run('pip install --quiet apache-beam')\n",
+    "\n",
+    "# Copy the input files into the local file system.\n",
+    "run('mkdir -p data')\n",
+    "run('gsutil cp gs://dataflow-samples/shakespeare/kinglear.txt data/kinglear.txt')\n",
+    "run('gsutil cp gs://apache-beam-samples/SMSSpamCollection/SMSSpamCollection data/SMSSpamCollection.csv')"
+   ],
+   "execution_count": null,
+   "outputs": []
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Inspect the data\n",
+    "We'll be working with 2 datasets. We'll use `kinglear.txt` for the first example - word count, and `SMSSpamCollection.csv` for the second and third.\n",
+    "Let's first take a loot at the `kinglear.txt` dataset."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "run('head data/kinglear.txt')"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "This is just a `txt` file - it contains lines of text.\n",
+    "Let's take a look at the other dataset."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "run('head data/SMSSpamCollection.csv')\n",
+    "run('wc -l data/SMSSpamCollection.csv')"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "This dataset is a `csv` file with 5,574 rows and 2 columns recording the following attributes separated by a tab sign:\n",
+    "1. `Column 1`: The label (either `ham` or `spam`)\n",
+    "2. `Column 2`: The SMS as raw text (type `string`)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## Example 1: word count\n",
+    "In this popular introductory exercise, we will build a pipeline that reads lines of text from the input dataset `kinglear.txt` and counts the number of times each word appears in the text.\n",
+    "To start, we'll create a `.yaml` file specifying our pipeline."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "pipeline = '''\n",
```

Comment:
Maybe do 

%%writefile pipeline-01.yaml

instead?

Reply:
Similarly elsewhere.

Reply:
I was considering this. I went for a custom function `save_to_file()` because Jupyter notebooks in some IDEs (like Intellij) will not highlight the text nicely when a `%%writefile` is used. Take a look at the screenshot:
![image](https://github.com/apache/beam/assets/7072475/643e838b-3abd-4997-8de7-112866ea084d)

On the other hand, I checked for the same in Google Colab and the highlighting there is absent, which makes it look better. 
![image](https://github.com/apache/beam/assets/7072475/6c3a87f3-e589-426c-9838-6cb2f669b2fe)

I'd still opt for my custom function, but let me know what you think.

Reply:
Raw jupyter will not highlight the `%%writefile` nor a String version, so I'd still opt for my solution, given it is better in Intellij and it doesn't matter for the raw Jupyter

Reply:
I'm not overly concerned with how it displays in IntelliJ (especially considering the target audience is not using an IDE anyway), and I think the lack of highlighting in raw Jupyter does not make it unclear what is going on, and makes editing the yaml easier.

For reference, this is how it looks in raw Jupyter
![image](https://github.com/apache/beam/assets/35542536/5aa9bb83-85f0-498b-9679-82d4941a11a2)

I would opt for the inline function - WDYT @robertwb